src/validation.ts

```diff
@@ -1,57 +1,97 @@
-import { AzureFunction, Context, HttpRequest } from '@azure/functions';
+import { HttpHandler, HttpRequest, InvocationContext } from '@azure/functions';
 import { AnySchema } from 'joi';
 
 import { ApplicationError } from './error';
+import { BeforeExecutionFunction, MiddlewareResult, PostExecutionFunction, isErrorResult } from './middleware';
+import { stringify } from './util/stringify';
 
 type ValidationOptions = Partial<{
     transformErrorMessage: (message: string) => unknown;
-    extractValidationContentFromRequest: (context: Context, req: HttpRequest) => unknown;
+    extractValidationContentFromRequest: (
+        req: HttpRequest,
+        context: InvocationContext,
+        result: MiddlewareResult<ReturnType<HttpHandler>>,
+    ) => unknown;
     shouldThrowOnValidationError: boolean;
+    skipIfResultIsFaulty: boolean;
+    printRequest: boolean;
 }>;
 
-export function requestValidation(schema: AnySchema, opts?: ValidationOptions): AzureFunction {
-    return (context: Context, req: HttpRequest): Promise<void> => {
-        context.log.verbose('Validating the request.');
+export function requestValidation(schema: AnySchema, opts?: ValidationOptions): BeforeExecutionFunction<HttpHandler> {
+    const skipIfResultIsFaulty = opts?.skipIfResultIsFaulty ?? true;
+    const printRequest = opts?.printRequest ?? false;
 
-        const toBeValidatedContent = opts?.extractValidationContentFromRequest?.(context, req) ?? req.body;
-        const shouldThrowOnValidationError = opts?.shouldThrowOnValidationError ?? true;
+    return async (req, context, result): Promise<void> => {
+        if (skipIfResultIsFaulty && result.$failed) {
```

Comment:
Müsste das nicht eher in die Response Validation?

Reply:
Warum?

Reply:
Ich dachte es geht hier darum, wenn das Result der eigentlichen Funktion faulty ist. Wenn ich es jetzt richtig verstanden habe, ist hier aber gemeint, dass ein zuvor ausgeführter before-Handler einen Fehler geworfen hat oder?

Reply:
Genau.